### PR TITLE
创建本地化文档自动翻译工具 & 更新本地化文档

### PR DESCRIPTION
**Pull request by UniMars**
@MistEO github ci回头我研究研究，先把这个合并一下，不然我怕别人提交后又要冲突了

### CODE REVIEW

**Comment by MistEO** [comment]
@ABA2396 这个 PR 怎么说啊（

File: src/MaaWpfGui/Res/Localizations/ja-jp.xaml [hunk 1]
```diff
@@ -222,14 +226,15 @@
...
+    <system:String x:Key="RefreshTraderWithDice">ダイスでトレーダーをリフレッシュ</system:String>
 
     <system:String x:Key="PenguinID">Penguin-StatsのレポートのID（デジタル部分のみ）</system:String>
     <system:String x:Key="DrGrandet">Dr.グランデモード</system:String>
     <system:String x:Key="DrGrandetTip">理性の回復確認画面で、現在の1ポイントの理性回復が完了するまで待ってから、すぐに確認してください。</system:String>
 
     <system:String x:Key="SocialPtShop">FP交換</system:String>
     <system:String x:Key="CreditFight">サポートを借りて戦闘を行いポイントを獲得します</system:String>
-    <system:String x:Key="CreditFightTip" xml:space="preserve">基地訪問後、OF-1でサポートを使用すると、30クレジット獲得できます。&#x0a;この機能は、ステージを現在/前回に選択している場合は無効です。</system:String>
+    <system:String x:Key="CreditFightTip" xml:space="preserve">基地訪問後、OF-1でサポートを使用すると、30クレジット獲得できます。\nこの機能は、ステージを現在/前回に選択している場合は無効です。</system:String>
```
Comment (MistEO) [inline src/MaaWpfGui/Res/Localizations/ja-jp.xaml:237]: 这种直接走 views 显示到界面上的文本（而非在 .cs 代码中读取的），\n 有试过可以正常换行吗？

**Comment by liuyifan-eric** [comment]
这个如何避免手动翻译的JP/KR被覆写掉？

File: src/MaaWpfGui/Res/Localizations/ja-jp.xaml (hunk 1 above)
Comment (MistEO) [reply src/MaaWpfGui/Res/Localizations/ja-jp.xaml:237]: WYE228MQ8M`QVRKJL%`{G`G_tmb

有问题

**Comment by UniMars** [comment]
> 这个如何避免手动翻译的JP/KR被覆写掉？

目前本地的翻译有两种模式，一种是只更新基准语言节点的增减和移动，对于共有的同键同位置的节点保持不动，
还有一种是调git获得基准语言的新旧版本，把更新情况全部同步过去，包括value的修改
github action还没写，但我初步的想法是用tag版本来标记

**Comment by UniMars** [comment]
@wallsman Is the fix all about polish up, or is threre any mistake that is different from the right meaning?

**Comment by wallsman** [comment]
> @wallsman Is the fix all about polish up, or is threre any mistake that is different from the right meaning?
I have corrected the parts I mistranslated in the past and the parts I found in this translation that I almost misinterpreted the meaning of.
In the meantime, I have tried to make the wording natural in Japanese.

自分が過去に誤訳していた部分と、今回の翻訳で意味を取り違えそうになった部分を発見したので修正しています。
一応、日本語での自然な言い回しにしています。

**Comment by UniMars** [comment]
> > @wallsman Is the fix all about polish up, or is threre any mistake that is different from the right meaning?
> 
> I have corrected the parts I mistranslated in the past and the parts I found in this translation that I almost misinterpreted the meaning of. In the meantime, I have tried to make the wording natural in Japanese.
> 
> 自分が過去に誤訳していた部分と、今回の翻訳で意味を取り違えそうになった部分を発見したので修正しています。 一応、日本語での自然な言い回しにしています。

I'm sorry, but i don't know Japanese. Actually, the translation is made by machine. i want to know that is there any errors that the sentence was translated to  another meaning compared to the right meaning. If there is, i need to modify my code. If it is all about expression and usage of words, i won't need to do that.

**Comment by wallsman** [comment]
> > > @wallsman Is the fix all about polish up, or is threre any mistake that is different from the right meaning?
> > 
> > 
> > I have corrected the parts I mistranslated in the past and the parts I found in this translation that I almost misinterpreted the meaning of. In the meantime, I have tried to make the wording natural in Japanese.
> > 自分が過去に誤訳していた部分と、今回の翻訳で意味を取り違えそうになった部分を発見したので修正しています。 一応、日本語での自然な言い回しにしています。
> 
> I'm sorry, but i don't know Japanese. Actually, the translation is made by machine. i want to know that is there any errors that the sentence was translated to another meaning compared to the right meaning. If there is, i need to modify my code that make the translation. If it is all about expression and usage of words, i won't need to do that.

Ah OK.
Fix Point.
><system:String x:Key="HelpUsWithOverseasServersTip">外国のサーバーの適応は不可能です、助けてください！プログラミングの知識は必要ありません。</system:String>
> ><system:String x:Key="HelpUsWithOverseasServersTip">グローバルサーバーの対応は非常に大変で、皆様の協力が必要です！プログラミングの知識は必要ありません。</system:String>

The meaning of "difficult" has become impossible to translate automatically.

Otherwise, no problems at all. It's just that it's hard to get the meaning across.

**Comment by UniMars** [comment]
> The meaning of "difficult" has become impossible to translate automatically.
> 
> Otherwise, no problems at all. It's just that it's hard to get the meaning across.
Glad to know there is only one translate mistake. BTW, because the translation was made by chatgpt instead of oher ordinary translation tools, what do you evaluate the language proficiency of this translation, compared with native speaker, 2nd language learner and google translate.

**Comment by wallsman** [comment]
> Glad to know there is only one translate mistake. BTW, because the translation was made by chatgpt instead of oher ordinary translation tools, how do you evaluate the language proficiency of this translation, compared with native speaker, 2nd language learner and google translate.

I use DeepL, but I am not bright enough in other languages to avoid evaluating it. Currently, DeepL is probably the closest to a natural translation, but since Google Translate is quite close to a direct translation, it tends to be unnatural and even changes the meaning at many times.

However, Japanese is a difficult language to translate, even if you are a native speaker and have been using it for many years, so it will take some time before any tool will produce the translation results as you expect.

**Comment by UniMars** [comment]
How do you score between deepl, google translation, this translation and native speaker

**Comment by wallsman** [comment]
> How do you score between deepl, google translation, this translation and native speaker

Google Translate 50 points (because it can be used)
DeepL 80 points (because it is easy to make sentences that make sense and can be used for business purposes)
ChatGPT (You) 80 points (because I don't feel any translation discrepancies at this stage)

**Comment by MistEO** [comment]
Thx-ld

**Comment by ABA2396** [comment]
这不正好测测配置文件的翻译）

**Pull request merged**